File: email-validation-3-dot-net-examples/ValidateEmailAddressRestSdkExample.cs
using System;
using System.Threading.Tasks;
using email_validation_3_dot_net.REST;

namespace email_validation_3_dot_net_examples
{
    public static class ValidateEmailAddressRestSdkExample
    {
        public static void Go(string licenseKey, bool isLive)
        {
            Console.WriteLine("\r\n-----------------------------------------------------------");
            Console.WriteLine("Email Validation 3 - ValidationEmailAddressInput - REST SDK");
            Console.WriteLine("-----------------------------------------------------------");

            ValidateEmailAddressClient.ValidationEmailAddressInput validationEmailAddressInput = new
            (
                EmailAddress: "Johan@gmail.com",
                LicenseKey: licenseKey,
                AllowCorrections: "true",
                IsLive: isLive,
                Timeout: "10000",
                TimeoutSeconds: 15
            );

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Email Address    : {validationEmailAddressInput.EmailAddress}");
            Console.WriteLine($"License Key      : {validationEmailAddressInput.LicenseKey}");
            Console.WriteLine($"Allow Corrections: {validationEmailAddressInput.AllowCorrections}");
            Console.WriteLine($"Is Live          : {validationEmailAddressInput.IsLive}");
            Console.WriteLine($"Timeout          : {validationEmailAddressInput.Timeout}");
            Console.WriteLine($"Timeout Seconds  : {validationEmailAddressInput.TimeoutSeconds}");

            EV3Response response = ValidateEmailAddressClient.Invoke(validationEmailAddressInput);
            if (response.Error is null && response.ValidateEmailInfo != null)
            {
                Console.WriteLine("\r\n* Email Validation Details *\r\n");

                Console.WriteLine($"Score                       : {response.ValidateEmailInfo.Score}");
                Console.WriteLine($"Is Deliverable              : {response.ValidateEmailInfo.IsDeliverable}");
                Console.WriteLine($"Email Address In            : {response.ValidateEmailInfo.EmailAddressIn}");
                Console.WriteLine($"Email Address Out           : {response.ValidateEmailInfo.EmailAddressOut}");
                Console.WriteLine($"Email Corrected             : {response.ValidateEmailInfo.EmailCorrected}");
                Console.WriteLine($"Box                         : {response.ValidateEmailInfo.Box}");
                Console.WriteLine($"Domain                      : {response.ValidateEmailInfo.Domain}");
                Console.WriteLine($"Top Level Domain            : {response.ValidateEmailInfo.TopLevelDomain}");
                Console.WriteLine($"Top Level Domain Description: {response.ValidateEmailInfo.TopLevelDomainDescription}");
                Console.WriteLine($"Is SMTP Server Good         : {response.ValidateEmailInfo.IsSMTPServerGood}");
                Console.WriteLine($"Is Catch All Domain         : {response.ValidateEmailInfo.IsCatchAllDomain}");
                Console.WriteLine($"Is SMTP Mailbox Good        : {response.ValidateEmailInfo.IsSMTPMailBoxGood}");
                Console.WriteLine($"Warning Codes               : {response.ValidateEmailInfo.WarningCodes}");
                Console.WriteLine($"Warning Descriptions        : {response.ValidateEmailInfo.WarningDescriptions}");
                Console.WriteLine($"Notes Codes                 : {response.ValidateEmailInfo.NotesCodes}");
                Console.WriteLine($"Notes Descriptions          : {response.ValidateEmailInfo.NotesDescriptions}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Type       : {response.Error?.Type}");
                Console.WriteLine($"Error Type Code  : {response.Error?.TypeCode}");
                Console.WriteLine($"Error Description: {response.Error?.Desc}");
                Console.WriteLine($"Error Desc Code  : {response.Error?.DescCode}");
            }
        }
    }
}

File: email-validation-3-python-examples/validate_email_address_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../email-validation-3-python/REST"))

from validate_email_address_rest import validate_email_address

def validate_email_address_rest_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n-----------------------------------------------------------");
    print("Email Validation 3 - ValidationEmailAddressInput - REST SDK");
    print("-----------------------------------------------------------");

    email_address = "jan@serviceobjects.com";
    allow_corrections = "true";
    timeout = "10000";
    timeout_seconds = 15;
 
    print(f"\r\n* Input *\r\n");
    print(f"Email Address    : {email_address}");
    print(f"License Key      : {license_key}");
    print(f"Allow Corrections: {allow_corrections}");
    print(f"Is Live          : {is_live}");
    print(f"Timeout          : {timeout}");
    print(f"Timeout Seconds  : {timeout_seconds}");

    try:
        response=validate_email_address(email_address, allow_corrections, license_key, timeout, is_live, timeout_seconds)

        if response.Error is None:
            print("\r\n* Email Validation Details *\r\n");
            print(f"Score                       : {response.ValidateEmailInfo.Score}");
            print(f"Is Deliverable              : {response.ValidateEmailInfo.IsDeliverable}");
            print(f"Email Address In            : {response.ValidateEmailInfo.EmailAddressIn}");
            print(f"Email Address Out           : {response.ValidateEmailInfo.EmailAddressOut}");
            print(f"Email Corrected             : {response.ValidateEmailInfo.EmailCorrected}");
            print(f"Box                         : {response.ValidateEmailInfo.Box}");
            print(f"Domain                      : {response.ValidateEmailInfo.Domain}");
            print(f"Top Level Domain            : {response.ValidateEmailInfo.TopLevelDomain}");
            print(f"Top Level Domain Description: {response.ValidateEmailInfo.TopLevelDomainDescription}");
            print(f"Is SMTP Server Good         : {response.ValidateEmailInfo.IsSMTPServerGood}");
            print(f"Is Catch All Domain         : {response.ValidateEmailInfo.IsCatchAllDomain}");
            print(f"Is SMTP Mailbox Good        : {response.ValidateEmailInfo.IsSMTPMailBoxGood}");
            print(f"Warning Codes               : {response.ValidateEmailInfo.WarningCodes}");
            print(f"Warning Descriptions        : {response.ValidateEmailInfo.WarningDescriptions}");
            print(f"Notes Codes                 : {response.ValidateEmailInfo.NotesCodes}");
            print(f"Notes Descriptions          : {response.ValidateEmailInfo.NotesDescriptions}");
        else:
            if hasattr(response, 'Error') and response.Error:
                print("\n* Error *\n")
                print(f"Error Type       : {response.Error.Type}")
                print(f"Error Type Code  : {response.Error.TypeCode}")
                print(f"Error Description: {response.Error.Desc}")
                print(f"Error Desc Code  : {response.Error.DescCode}")

    except Exception as e:
        print(f"\nException occurred: {str(e)}")
